src/Runtime/OMTensor.inc

```diff
@@ -423,63 +423,71 @@ void omTensorPrint(const char *msg, const OMTensor *tensor) {
   printf("\trank = %lld\n", (long long)rank);
   printf("\tdataType = %s\n", getDataTypeName(dataType));
   printf("\tnumElems = %lld\n", (long long)omTensorGetNumElems(tensor));
+  printf("\tshape: ");
+  for (int64_t i = 0; i < rank; i++)
+    printf("[%lld]", (long long)shape[i]);
+  printf("\n");
   printf("\tstrides: ");
   for (int64_t i = 0; i < rank; i++)
     printf("[%lld]", (long long)strides[i]);
   printf("\n");
 
-  printf("\tdata: ([");
+#define START_LOOP(IV, LB, UB)                                                 \
```

Comment:
My personal preference is not to go crazy with macros. It would feel that using a lambda function here would be perfectly appropriate.

If you really want, you can even do recursive lambda functions where you expand a `{I, J, K}, {ilb, jlb, akb}, {iub, jub, kub}` into a 3 level lambda that expires exactly what you want.

Reply:
Understand your dislike of macros. I used macros here because this is a C function, part of libcruntime.a and so can't use C++ here.  I looked and currently the compiler can only generate code to C function defined in the runtime, because `onnx-mlir` links only libcruntme.a which contains exclusively objects with C source code.

Reply:
Got it, that makes sense. For the record, I don't dislike macros (I used to use quite vile recursive macros...) but find C++ constructs more readable. It make sense to use macro in this context.

Reply:
Using variadic macros now.